lib/sdk/io/file.js

```diff
@@ -129,20 +129,30 @@ exports.list = function list(path) {
 };
 
 exports.open = function open(filename, mode) {
-  var file = MozFile(filename);
-  if (typeof(mode) !== "string")
+  const file = MozFile(filename);
+  if (typeof (mode) !== "string") {
     mode = "";
+  }
 
   // File opened for write only.
-  if (/w/.test(mode)) {
-    if (file.exists())
+  if (/[aw]/.test(mode)) {
+    const exists = file.exists();
+    if (exists) {
       ensureFile(file);
-    var stream = Cc['@mozilla.org/network/file-output-stream;1'].
-                 createInstance(Ci.nsIFileOutputStream);
-    var openFlags = OPEN_FLAGS.WRONLY |
-                    OPEN_FLAGS.CREATE_FILE |
-                    OPEN_FLAGS.TRUNCATE;
-    var permFlags = parseInt("0644", 8); // u+rw go+r
+    }
+    const stream = Cc['@mozilla.org/network/file-output-stream;1'].
+	  createInstance(Ci.nsIFileOutputStream);
+    let openFlags = OPEN_FLAGS.WRONLY;
+
+    if (exists && /a/.test(mode)) {
+      openFlags |= OPEN_FLAGS.APPEND;
+    }
+    else {
+      openFlags |= OPEN_FLAGS.TRUNCATE |
+        OPEN_FLAGS.CREATE_FILE;
```

Comment:
You can include CREATE_FILE regardless of whether the file exists or not. If the file already exists it has no effect.